version: '3.9'

services:
  futuradb:
    image: postgres
    container_name: futuradb
    restart: always
    command:
      - "postgres"
      - "-c"
      - "max_prepared_transactions=64"
    environment:
      - POSTGRES_USER=postgres
      - POSTGRES_PASSWORD=admin
    ports:
      - "5432:5432"
    volumes:
      - ./InitScripts/Postgress:/docker-entrypoint-initdb.d/docker_postgres_init.sql
      - "../../Volumes/postgres:/var/lib/postgresql/data"
    networks:
      - postgres-network

  pgadmin:
    image: dpage/pgadmin4
    environment:
      PGADMIN_DEFAULT_EMAIL: "futuraadmin@futurasistemas.com.br"
      PGADMIN_DEFAULT_PASSWORD: "futura1313"
    ports:
      - "16543:80"
    volumes:
      - pgadmin_data:/root/.pgadmin
    networks:
      - postgres-network
    depends_on:
      - futuradb
    
  rabbitmq:
    image: rabbitmq:3-management-alpine
    ports:
      - "15672:15672"
      - "5672:5672"

  
networks:
  postgres-network:
    driver: bridge

volumes:
  pgadmin_data:
  
  logs-folder:
        name: ${log_rabbitmq_management}
        driver: local
